npm/design-system/src/components/FileExplorer/FileExplorerItem.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { ReactNode } from 'react'
+import { InlineIcon } from '@iconify/react'
+import javascriptIcon from '@iconify/icons-vscode-icons/file-type-js-official'
+import typescriptIcon from '@iconify/icons-vscode-icons/file-type-typescript-official'
+import reactJs from '@iconify/icons-vscode-icons/file-type-reactjs'
+import reactTs from '@iconify/icons-vscode-icons/file-type-reactts'
+import folderClosed from '@iconify/icons-vscode-icons/default-folder'
+import folderOpen from '@iconify/icons-vscode-icons/default-folder-opened'
+import { FolderOrFile } from './types'
+import styles from './FileExplorer.module.scss'
+import cs from 'classnames'
+
+const icons: Record<string, any> = {
+  js: { icon: javascriptIcon },
+  ts: { icon: typescriptIcon },
+  tsx: { icon: reactTs },
+  jsx: { icon: reactJs },
+  folderOpen: { icon: folderOpen },
+  folderClosed: { icon: folderClosed },
+}
+
+const getExt = (path: string) => {
+  const extensionMatches = path.match(/(?:\.([^.]+))?$/)
+
+  return extensionMatches ? extensionMatches[1] : ''
+}
+
+export interface FileExplorerItemProps {
+  item: FolderOrFile
+  children?: ReactNode
+  style?: React.CSSProperties
+  getHref? (item: FolderOrFile): string
+  depth: number
+  onFocus: any
+  onClick: any
+  onBlur: any
+}
+
+export const FileExplorerItem: React.FC<FileExplorerItemProps> = (props) => {
+  const ext = getExt(props.item.shortName) || ''
+  const folderIcon = props.item.isOpen ? icons.folderOpen : icons.folderClosed
+  const getHref = props.getHref || (() => '#')
+  const onClick = props.item.onClick || (() => {})
+
+  const inlineIconProps = ext ? icons[ext] : folderIcon
+
+  // Negative margins let the <a> tag take full width (a11y)
+  // while the <li> tag with text content can be positioned relatively
+  // This gives us HTML + CSS-only highlight and click handling
+  const inlineStyles = {
+    a: {
+      marginLeft: `calc(-20px * ${props.depth})`,
+      width: `calc(100% + (20px * ${props.depth}))`,
+    },
+    li: {
+      marginLeft: `calc(20px * ${props.depth})`,
+    },
+  }
+
+  return (
+    <a
```

Comment:
Please use `<button` for a11y reasons

Reply:
Isn't <a> better than button for hash navigation changes?

Reply:
You are passing both `onClick` and `href` which is really weird. I don't understand this – it looks like you are trying to sync url and state which is a problem all the time. We need to have a single source of trust. 

So if you want to use url – you need to subscribe to url changes and use hash without any internal state

Reply:
`href` is fine as long as `onClick` is calling `preventDefault`. This is the "correct" way to build buttons that perform as links, so that you still get standard browser right/middle click behavior.

Reply:
Yes, but it is unclear that it is doing prevent default. Maybe not, how would you know from this code? If it is – we need some kind of `<Link />` component. But we don't really have any kind of routing, so it looks confusing

Reply:
Agreed